dhcpv4/nclient4/client.go

```diff
@@ -627,6 +628,29 @@ func (c *Client) SendAndRead(ctx context.Context, dest *net.UDPAddr, p *dhcpv4.D
 	return response, nil
 }
 
+// TransactAckNak exposes a higher-level API than SendAndRead for sending a given request (any type)
+// and receiving an ACK/NAK response. The server identifier is validated, responses are type checked,
+// and NAK responses are automatically returned as errors.
+func (c *Client) TransactAckNak(ctx context.Context, request *dhcpv4.DHCPv4, serverIdentifier net.IP) (*dhcpv4.DHCPv4, error) {
```

Comment:
Does this need to be a public function?

Reply:
Not necessarily, I've changed it to be private.